akka-actor/src/main/scala/akka/routing/DefaultMetricsBasedResizer.scala

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.routing
+
+import java.time.{ Duration, LocalDateTime }
+
+import scala.collection.immutable
+import scala.concurrent.forkjoin.ThreadLocalRandom
+import com.typesafe.config.Config
+
+import akka.actor._
+
+import DefaultMetricsBasedResizer._
+
+trait MetricsBasedResizer extends Resizer {
```

Comment:
I would like to find a better, more specific, name for this. Metrics is so general.
What does it use? mailbox size? throughput?
Perhaps something with Adaptive or Optimizing in the name?
Sorry that I don't have any good suggestions.